p4include/v1model.p4

```diff
@@ -537,43 +528,65 @@ extern void verify_checksum_with_payload<T, O>(in bool condition, in T data, in
 extern void update_checksum_with_payload<T, O>(in bool condition, in T data, inout O checksum, HashAlgorithm algo);
 
 /***
- * Calling resubmit during execution of the ingress control will,
- * under certain documented conditions, cause the packet to be
+ * clone is in most ways identical to the clone_preserving_field_list operation, with the
+ * only difference being that it never preserves any user-defined
+ * metadata fields with the cloned packet.  It is equivalent to
+ * calling clone_preserving_field_list with the same type and session parameter values,
+ * with empty data.
+ */
+extern void clone(in CloneType type, in bit<32> session);
+
+@deprecated("Please use 'resubmit_preserving_field_list' instead")
+extern void resubmit<T>(in T data);
+/***
+ * Calling resubmit_preserving_field_list during execution of the ingress
+ * control will cause the packet to be
  * resubmitted, i.e. it will begin processing again with the parser,
  * with the contents of the packet exactly as they were when it last
  * began parsing.  The only difference is in the value of the
  * standard_metadata instance_type field, and any user-defined
  * metadata fields that the resubmit operation causes to be
  * preserved.
  *
- * The value of the user-defined metadata fields that are preserved in
- * resubmitted packets is the value they have at the end of ingress
- * processing, not their values at the time the resubmit call is made.
- * See Note 2 for issues with this.
+ * The user metadata fields that are tagged with @field_list(index) will
+ * be sent to the parser together with the packet.
  *
  * Calling resubmit is only supported in the ingress control.  There
  * is no way to undo its effects once it has been called.  If resubmit
  * is called multiple times during a single execution of the ingress
  * control, only one packet is resubmitted, and only the data from the
  * last such call is preserved.  See the v1model architecture
  * documentation (Note 1) for more details.
+ *
+ * For example, the user metadata fields can be annotated as follows:
+ * struct UM {
+ *    @field_list(1)
+ *    bit<32> x;
+ *    @field_list(1, 2)
+ *    bit<32> y;
+ *    bit<32> z;
+ * }
```

Comment:
Thanks for adding a short example to this documentation!

Reply:
I created a separate issue to remind me to either myself, or find some other volunteer to help, update other documentation inside and outside of the p4lang/p4c repository on how these operations work.  I think it is best of those changes are made in their own separate PRs after this one is merged: https://github.com/p4lang/p4c/issues/2977